source/common/router/router_ratelimit.cc

```diff
@@ -109,7 +109,15 @@ bool RemoteAddressAction::populateDescriptor(RateLimit::DescriptorEntry& descrip
     return false;
   }
 
-  descriptor_entry = {"remote_address", remote_address->ip()->addressAsString()};
+  if (prefix_len_ == RemoteAddressAction::PREFIX_LEN_UNSET) {
+    descriptor_entry = {"remote_address", remote_address->ip()->addressAsString()};
+    return true;
+  }
+
+  Network::Address::CidrRange cidr_entry =
+      Network::Address::CidrRange::create(remote_address->ip()->addressAsString(), prefix_len_);
```

Comment:
In case of the `prefix_len` == 0, then it will get an any-address https://github.com/envoyproxy/envoy/blob/52455a260c0d1b4c6cb59aa23c290a28412794ad/source/common/network/cidr_range.cc#L154

Is this the case we expect? maybe we should limit the `prefix_len`'s value to greater than 0.